### PR TITLE
Migrate Pagination to TypeScript

### PR DESCRIPTION
**Pull request by VanAnderson**
This PR migrates the `Pagination` component to TypeScript as part of the [TypeScript refactor](https://github.com/primer/components/issues/970).

### CODE REVIEW

**Comment by VanAnderson** [comment]
forgot to export PaginationProps! done in 43c79d6

**Comment by VanAnderson** [comment]
8fb049d should fix the last failing TS error 🤞

File: src/Pagination/Pagination.tsx [hunk 1]
```diff
@@ -146,17 +158,28 @@ const PaginationContainer = styled.nav`
   ${sx};
 `
 
+type PaginationParameters = {
```
Comment (colebemis) [inline src/Pagination/Pagination.tsx:161]:
```suggestion
export type PaginationProps = {
```

File: src/Pagination/Pagination.tsx [hunk 2]
```diff
@@ -203,4 +226,5 @@ Pagination.defaultProps = {
   theme
 }
 
+export type PaginationProps = ComponentProps<typeof Pagination>
```
Comment (colebemis) [inline src/Pagination/Pagination.tsx:229]:
```suggestion
```

**Review by colebemis — APPROVED** [review]
Awesome 👍  Just left one suggestion about the PaginationProps type

**Pull request merged**